wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -0,0 +1,176 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import SwiftUI
+import WireCommonComponents
+import WireDataModel
+import WireSyncEngine
+
+protocol DeviceDetailsViewActions {
+    func fetchCertificate() async -> E2eIdentityCertificate?
+    func showCertificate(_ certificate: String)
+    func removeDevice() async -> Bool
+    func resetSession() async -> Bool
+    func updateVerified(_ value: Bool) async -> Bool
+    var isProcessing: ((Bool) -> Void)? { get set }
+    func copyToClipboard(_ value: String)
+}
+
+final class DeviceInfoViewModel: ObservableObject {
+    let userSession: UserSession
+    let uuid: String
+    let addedDate: String
+    let proteusID: String
+    let getUserClientFingerprintUseCase: GetUserClientFingerprintUseCaseProtocol
```

Comment:
as convention, we drop usecase suffix for the property name
```suggestion
    let getUserClientFingerprint: GetUserClientFingerprintUseCaseProtocol
```

Reply:
done